### PR TITLE
switch to go modules

### LINKED ISSUE #2451: Switch to using go modules
[Go modules](https://github.com/golang/go/wiki/Modules) are nicer than `dep` for handling dependency management. We should switch skaffold over to use them.

### PR DESCRIPTION
**Pull request by nkubala**
only a few real dependency changes here:
* bump k8s client libraries from 1.12.9 to 1.12.10. these removed some deprecated options, including `IncludeUninitialized` for pod selectors. we'll need to monitor the health check and loggers to make sure this doesn't have any unintended side effects.
* bump cloudbuild client library from 0.17.0 to 0.43.0. surprisingly not a big change for skaffold, `cloudbuild.New(google.Client)` was deprecated in favor of `cloudbuild.NewService(ctx)` which results in slightly less overhead on our side.

steps taken to generate this `go.mod`:
1) clone skaffold into a separate directory NOT in $GOPATH
2) `go mod init`
3) `go mod edit -replace=gopkg.in/russross/blackfriday.v2@v2.0.1=github.com/russross/blackfriday/v2@v2.0.1`
4) replace `github.com/google/go-github v0.0.0-20180926004559-f55b50f38167` with `github.com/google/go-github v17.0.0+incompatible` (manual)
5) replace `cloud.google.com/go v0.17.0` with `cloud.google.com/go v0.43.0` (manual)
6) `go mod tidy`

once this gets merged, I'll submit a follow up PR to upgrade dependencies.

Fixes #2451

### CODE REVIEW

**Comment by corneliusweig** [comment]
@nkubala Did you intentionally not use a `GOPROXY`? https://proxy.golang.org/

This could be enabled by adding this in the travis environment variables:
```yaml
envs:
- GOPROXY=https://proxy.golang.org
```

File: .travis.yml [hunk 1]
```diff
@@ -6,11 +6,19 @@ go: "1.12.x"
 git:
   submodules: false
 
+env:
+  global:
+    - GO111MODULE=on
```
Comment (dgageot) [inline .travis.yml]: I think this should be set in the Makefile. I had to run `GO111MODULE=on make`

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,79 @@
+module github.com/GoogleContainerTools/skaffold
```
Comment (dgageot) [inline go.mod]: I ran `GO111MODULE=on go mod tidy` and `GO111MODULE=on go mod vendor` and it removed a lot of dependencies.

File: hack/test.sh [hunk 3]
```diff
@@ -33,7 +33,6 @@ scripts=(
     "hack/boilerplate.sh"
     "hack/gofmt.sh"
     "hack/linter.sh"
-    "hack/dep.sh"
```
Comment (dgageot) [inline hack/test.sh]: Shouldn't we check that all deps are vendored?

**Comment by dgageot** [comment]
@nkubala another thing: I wonder if downloading the deps from the golang proxy wouldn't be faster that restoring the cache.
I ran `GOPROXY=https://proxy.golang.org GO111MODULE=on go mod vendor` on my machine with a clean cache and it takes about 12s

**Comment by dgageot** [comment]
The work has continued here: https://github.com/GoogleContainerTools/skaffold/pull/2541

**Pull request closed without merging**